ml-algorithms/src/main/java/org/opensearch/ml/engine/algorithms/metrics_correlation/MetricsCorrelation.java

```diff
@@ -0,0 +1,272 @@
+                Map<String, Object> modelInfo = searchHits[0].getSourceAsMap();
+                listener.onResponse(modelInfo);
+            }
+        }, e -> {
+            log.error("Failed to update model state", e);
+            listener.onFailure(e);
+        }));
+    }
+
+    public void registerModel(ActionListener<MLRegisterModelResponse> listener) throws InterruptedException {
+
+        FunctionName functionName = FunctionName.METRICS_CORRELATION;
+        MLModelFormat modelFormat = MLModelFormat.TORCH_SCRIPT;
+        String modelType = "custom";
+        MLModelConfig modelConfig = MetricsCorrelationModelConfig.builder()
+                .modelType(modelType)
+                .allConfig(null).build();
+        MLRegisterModelInput input = MLRegisterModelInput
+                .builder()
+                .functionName(functionName)
+                .modelName(MODEL_NAME)
+                .version(MCORR_ML_VERSION)
+                .modelFormat(modelFormat)
+                .modelConfig(modelConfig)
+                .url(MCORR_MODEL_URL)
+                .deployModel(true)
+                .build();
+        MLRegisterModelRequest registerRequest = MLRegisterModelRequest.builder().registerModelInput(input).build();
+
+        client.execute(MLRegisterModelAction.INSTANCE, registerRequest, ActionListener.wrap(mlDeployModelResponse -> {
+            listener.onResponse(mlDeployModelResponse);
+        }, e -> {
+            log.error("Failed to Register Model", e);
+            listener.onFailure(e);
+        }));
+    }
+
+    private void deployModel(MLDeployModelRequest mlDeployModelRequest, ActionListener<MLDeployModelResponse> listener) {
+        client.execute(MLDeployModelAction.INSTANCE, mlDeployModelRequest, ActionListener.wrap(mlDeployModelResponse -> {
+            listener.onResponse(mlDeployModelResponse);
+        }, e -> {
+            log.error("Failed to deploy Model", e);
+            listener.onFailure(e);
+        }));
+    }
+
+    public float[][] processedInput(List<float[]> input) {
+        float[][] processInput = new float[input.size()][];
+        for (int i = 0; i < input.size(); i++) {
+            float[] innerList = input.get(i);
+            float[] temp = processInput[i] = new float[innerList.length];
+            System.arraycopy(innerList, 0, temp, 0, temp.length);
+        }
+        return processInput;
+    }
+
+    @Override
+    public MetricsCorrelationTranslator getTranslator() {
+        return new MetricsCorrelationTranslator();
+    }
+
+    public SearchRequest getSearchRequest() {
+        SearchSourceBuilder searchSourceBuilder = new SearchSourceBuilder();
+        searchSourceBuilder.fetchSource(new String[] { MLModel.MODEL_ID_FIELD,
+                        MLModel.MODEL_NAME_FIELD, MLModel.MODEL_STATE_FIELD, MLModel.MODEL_VERSION_FIELD },
+                new String[] { MLModel.OLD_MODEL_CONTENT_FIELD, MLModel.MODEL_CONTENT_FIELD });
```

Comment:
As this feature doesn't have old models, you can just use `MLModel.MODEL_CONTENT_FIELD`